src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,55 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = new Object[SIZE];
+        this.values = new Object[SIZE];
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndex(key);
+        if (index == -1) {
+            if (size == SIZE) {
```

Comment:
Such naming will be more readable
```suggestion
            if (size == MAX_SIZE) {
```